lib/build_targets/plugins.dart

```diff
@@ -64,151 +64,206 @@ class NativePlugins extends Target {
       ..createSync(recursive: true);
 
-    final List<String> userIncludes = <String>[];
-    final List<String> userSources = <String>[];
+    final BuildMode buildMode = buildInfo.buildInfo.mode;
+    final String buildConfig = buildMode.isPrecompiled ? 'Release' : 'Debug';
+    final Directory engineDir =
+        getEngineArtifactsDirectory(buildInfo.targetArch, buildMode);
+    final Directory commonDir = engineDir.parent.childDirectory('tizen-common');
+
+    final Directory clientWrapperDir =
+        commonDir.childDirectory('cpp_client_wrapper');
+    final Directory publicDir = commonDir.childDirectory('public');
+    clientWrapperDir
+        .listSync(recursive: true)
+        .whereType<File>()
+        .forEach(inputs.add);
+    publicDir.listSync(recursive: true).whereType<File>().forEach(inputs.add);
+
+    assert(tizenSdk != null);
+    final TizenManifest tizenManifest =
+        TizenManifest.parseFromXml(tizenProject.manifestFile);
+    final String profile = tizenManifest.profile;
+    final String apiVersion = tizenManifest.apiVersion;
+    final Rootstrap rootstrap = tizenSdk.getFlutterRootstrap(
+      profile: profile,
+      apiVersion: apiVersion,
+      arch: buildInfo.targetArch,
+    );
+    inputs.add(tizenProject.manifestFile);
+
     final List<String> userLibs = <String>[];
+    final List<String> pluginClasses = <String>[];
 
     for (final TizenPlugin plugin in nativePlugins) {
+      // Create a copy of the plugin to allow editing its projectFile.
+      final TizenPlugin pluginCopy = plugin.copyWith(
+        directory: environment.fileSystem.systemTempDirectory.createTempSync(),
+      );
+      copyDirectory(plugin.directory, pluginCopy.directory);
+
+      final List<String> properties = <String>[];
+      for (String line in plugin.projectFile.readAsLinesSync()) {
+        if (line.startsWith('type =')) {
+          line = 'type = staticLib';
+        }
+        properties.add(line);
+      }
+      pluginCopy.projectFile.writeAsStringSync(properties.join('\n'));
       inputs.add(plugin.projectFile);
 
-      // TODO(swift-kim): Currently only checks for USER_INC_DIRS, USER_SRCS,
-      // and USER_LIBS. More properties may be parsed in the future.
-      userIncludes.addAll(plugin.getPropertyAsAbsolutePaths('USER_INC_DIRS'));
-      userSources.addAll(plugin.getPropertyAsAbsolutePaths('USER_SRCS'));
+      final Map<String, String> variables = <String, String>{
+        'PATH': getDefaultPathVariable(),
+      };
+      final List<String> extraOptions = <String>[
+        '-fPIC',
+        '-I"${clientWrapperDir.childDirectory('include').path.toPosixPath()}"',
+        '-I"${publicDir.path.toPosixPath()}"',
+        '-D${buildInfo.deviceProfile.toUpperCase()}_PROFILE',
+      ];
 
-      final Directory headerDir = plugin.directory.childDirectory('inc');
-      if (headerDir.existsSync()) {
-        headerDir
+      final RunResult result = await _processUtils.run(<String>[
```

Comment:
How about adding `buildNative()` method to `TizenSdk`?
```dart
Future<File> buildNative(
    TizenBuildInfo buildInfo,
    Rootstrap rootstrap, 
    Directory buildDirectory,
    String expectedOutputPath, { 
    List<String> extras,
    Map<String, String> environment,
    }) async {
      ....
```
this part can be more simple like below:
```dart
     final File libFile = await tizenSdk.buildNative(
        buildInfo,
        rootstrap,
        pluginCopy.directory,
        'lib$libName.a',
        extras: <String>[
          '-fPIC',
          '-I"${clientWrapperDir.childDirectory('include').path.toPosixPath()}"',
          '-I"${publicDir.path.toPosixPath()}"',
          '-D${buildInfo.deviceProfile.toUpperCase()}_PROFILE',
        ],
        environment: <String, String>{
          'PATH': getDefaultPathVariable(),
        },
      );
```

Reply:
Thanks! I applied the suggestion but modified the method signature a bit to match the actual parameters of the `tizen build-native` command.

Reply:
I forgot to apply this to the NativeTpk target in `package.dart`. I'll update it as well.